test/rails4_dummy/config/application.rb

```diff
@@ -4,14 +4,11 @@
 
 require 'rails'
 require 'action_controller/railtie'
+require 'coverband'
+Coverband.eager_loading_coverage { Bundler.require(*Rails.groups) }
```

Comment:
We could remove the need for this line by patching Bundler.require. I think we would still need an early require of coverband within rails application.rb though since coverband has to be loaded before `Bundler.require` is called.

Reply:
hmm so with this method we would as part of installation require folks to modify the application.rb...

Kinda of hurts the zero config railties story...

So the issue we are trying to solve is https://github.com/danmayer/coverband/issues/261

Thinking a bit more of other ways to address still feels like we just aren't covering all cases related to initialization... 

Kind of feels like each solution has had a slightly different bug story... We started with 

`config.after_initialize` and `before_initialize` which I think would work for the bundler require command... is some combination of eager_load, and the initializer hooks something that could cover all scenarios... A case were more detailed bug reports on each of the issues would have been good.... Rake was disabling eager loading... the prepend solved that... After initialize was adding code we didn't want to load time vs runtime... That was avoided by ignoring first request... 

Anyways, just feel like we are still missing how to be cover all cases. I guess this PR works, but I don't love the solution.

Reply:
> Kinda of hurts the zero config railties story...

Agreed. I'm not happy about that either. I'm just not sure how we cleanly capture `Bundle.require` without inserting some code before the `Bundle.require`. The problem is that the `Bundle.require` itself is what requires the coverband library. We are tracking the code that requires the tracking code.

> Kind of feels like each solution has had a slightly different bug story...

Agreed that we are kind of playing whack a mole with each solution. I think that it's important that going forward we put the extra effort in to have a test case to reproduce each issue. Then we will evolve toward the best solution that makes all the tests pass.

> I guess this PR works, but I don't love the solution.

This PR is really only an example of how one could track runtime/eager_loading coverage of gems. It also doesn't really change the coverband library itself other than adding the [Coverband#eager_loading](https://github.com/danmayer/coverband/pull/265/files#diff-181f764c69748a3ceb0cc64e4a37ebaeR37) method taking a block which I kind of like. Maybe we can evolve to a place where we don't need custom code for tracking runtime/eager_loading coverage of gems but I think this takes us in a positive direction.

Reply:
Added a [commit](https://github.com/danmayer/coverband/pull/265/commits/4bca5e51ffc981bc66bc626b380c5fb64c5b9669) that moves most custom code out of application. Now all that's left for tracking `Bundle#require` of gems accurately is requiring coverband before the call to `Bundle#require`. This makes sense to me. Another thought is that coverband still works out of box with rails, but if you would like to track the require of gems, coverband must be loaded before hand.

Reply:
> Agreed that we are kind of playing whack a mole with each solution. I think that it's important that going forward we put the extra effort in to have a test case to reproduce each issue. Then we will evolve toward the best solution that makes all the tests pass.

yep agreed, some of these are hard to reproduce in tests, but I think with the new forked tests model we should be able to figure out a way.